oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java

```diff
@@ -237,33 +274,97 @@ protected GoogleCredentials() {
     this(new Builder());
   }
 
+  /**
+   * Constructor with an explicit access token and quotaProjectId.
+   *
+   * <p>Deprecated, please use the {@link GoogleCredentials#GoogleCredentials(Builder)} constructor
+   * whenever possible.
+   *
+   * @param accessToken initial or temporary access token
+   * @param quotaProjectId a quotaProjectId, a project id to be used for billing purposes
+   */
+  @Deprecated
   protected GoogleCredentials(AccessToken accessToken, String quotaProjectId) {
     super(accessToken);
     this.quotaProjectId = quotaProjectId;
+    this.universeDomain = Credentials.GOOGLE_DEFAULT_UNIVERSE;
   }
 
   /**
    * Constructor with explicit access token.
    *
    * @param accessToken initial or temporary access token
    */
+  @Deprecated
   public GoogleCredentials(AccessToken accessToken) {
     this(accessToken, null);
   }
 
+  /**
+   * Constructor that relies on a {@link GoogleCredential.Builder} to provide all the necessary
+   * field values for initialization.
+   *
+   * @param builder an instance of a builder
+   */
   protected GoogleCredentials(Builder builder) {
-    this(builder.getAccessToken(), builder.getQuotaProjectId());
+    super(builder.getAccessToken());
+    this.quotaProjectId = builder.getQuotaProjectId();
+
+    if (builder.universeDomain == null || builder.universeDomain.trim().isEmpty()) {
```

Comment:
Can be simplified:

```
this.universeDomain = builder.getUniverseDomain();
if (this.universeDomain == null || this.universeDomain.trim().isEmpty()) {
  this.universeDomain = Credentials.GOOGLE_DEFAULT_UNIVERSE;
}

Reply:
ah nope... finals can be assigned once and only once ) could have worked in c# though ))